### DISCUSSION TITLE
3.1.0rc1

### DISCUSSION THREAD

**Discussion by adehad** [pycontribs/jira#1194] [General]
Thank you all for your contributions and patience awaiting this pre-release. Although this is classed as a "pre-release" this is the most well tested this library has been in its entire history ! [Our coverage is just shy of 58%](https://app.codecov.io/gh/pycontribs/jira/). The reasoning behind classing this as a pre-release is that we changed our CI system and want to make sure that any regressions can be addressed before a "proper" release.

## Minor Changes

* Create token auth option (#991) @skaiaa
* add support for ranking after (#1160) @durka
* add PermissionScheme Resource (#1139) @Steffen223
*  Allow the adding of headers rather than just overwriting all headers (#1085) @adehad
* Make search_assignable_users_for_issues function GDPR compliant. (#1117) @nelli-acc
* Add expand to JIRA.project and JIRA.projects (#865) @joshheinrichs
* ResilientSession: do not log request data on ConnectionError (#928) @Flupp
* fix pickling and hashability (#1051) @adehad
* Allow expands for retrieval of comments (#1003) @matthewkeenan
* Close the file descriptor for add_attachment (#957) @yen3
* Adopt fstrings (flynt) (#1044) @studioj
* Improve feedback for oauth errors. (#888) @aklajnert
* Correct lookup of issuetypes when calling create_issue (#978) @soerenbe
* add typehints and allow google docstrings (#1023) @adehad

## Bugfixes

* Export resource_class_map and fix bug in Role.add_user() (#1047) @wlupton
* Jira Cloud user support (#1109) @adehad
* allow testing of Jira Cloud (#1107) @adehad
* Client Certificate Authentication for All Auth Types (#1191) @kchason
* Allow _get_user_accountid to find "Unassigned" and "Automatic" users. (#1152) @adehad
* use "agile" endpoint over now removed "greenhopper" (#1186) @adehad
* fix: rename default branch to main (#1166) @ssbarnea
* Fix JWT QSH generation for urls with repeated parameters (#1157) @robinhughes
* fix TB on RequestType instanciation with raw content (fixes #1130) (#1133) @bmathieu33
* Make search_assignable_users_for_issues function GDPR compliant. (#1117) @nelli-acc
* use requests.structures.CaseInsensitiveDict directly (#1084) @adehad
* Fix broken tests (#1079) @manc-lx
* Add (some) missing type hints for `_IssueFields` (#1063) @julenpardo
* Refactor locking of test dependencies (#1074) @ssbarnea
* fix assign issue behaviour and correct watcher functions docstrings (#1053) @adehad
* Allow you to set the reporter as well, not just the assignee (#987) @tlianza
* Close the file descriptor for add_attachment (#957) @yen3
* Make user search GDPR compliant (#927) @julenpardo
* fix: group_members (#933) @nilleb
* add typehints and allow google docstrings (#1023) @adehad
* Fix docstring generation of jira client and minor doc layouting fix (#1038) @studioj
* adding version to documentation to help out on #1016 (#1039) @studioj

Kudos goes to: @Flupp, @RonnyPfannschmidt, @Steffen223, @adehad, @aklajnert, @bmathieu33, @dependabot, @dependabot[bot], @durka, @joshheinrichs, @julenpardo, @kchason, @manc-lx, @matthewkeenan, @nelli-acc, @nilleb, @pre-commit-ci, @pre-commit-ci[bot], @robinhughes, @skaiaa, @soerenbe, @ssbarnea, @studioj, @tlianza, @tmct, @wlupton and @yen3

<hr /><em>This discussion was created from the release <a href='https://github.com/pycontribs/jira/releases/tag/3.1.0rc1'>3.1.0rc1</a>.</em>

**Comment by cheauailin** [comment]
Hi, I was wondering if i get an estimate of when this be released? :)

Thanks!

**Reply by adehad** [reply]
You can give it a whirl already using: `pip install jira==3.1.0rc1` !

For a specific date on a non-pre-release, I can't be certain at this stage, there is a regression reported in #1164 that I want to see if it can be address beforehand. If there are no more reported regressions for a week I'd be happier making another release.

**Reply by cheauailin** [reply]
Yes that would be awesome! My team would truly appreciate the new release as we are moving towards token auth. Would it be alright to keep us updated on when you'll be making that release? :) 

Thank you so much!

**Reply by adehad** [reply]
Yep, can drop a message here. Also consider "watching" for new releases (Watch > Custom > Releases):
![image](https://user-images.githubusercontent.com/26027314/140318239-39578bc9-45ad-4fc3-9692-a38a96f2e1b1.png)

**Reply by adehad** [reply]
`3.1.1` is now up !

FYI 
@cheauailin
